### ISSUE TITLE
[help] how to move emoji selection/command palette selection by tab/up/down/left/right key on windows

### ISSUE THREAD

**Issue by lost22git** [wez/wezterm#3644]
### What Operating System(s) are you seeing this problem on?

Windows

### WezTerm version

wezterm 20230427-063511-9bc7de70

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

[help] how to move emoji selection/command palette selection by up/down/left/right key on windows

![cc](https://user-images.githubusercontent.com/65008815/235316398-469b2c03-496e-48b5-bb40-2efea9a64ba0.gif)

### Configuration

```lua
-- Dump any object to string
local function dump(o)
  if type(o) == 'table' then
    local s = '{ '
    for k, v in pairs(o) do
      if type(k) ~= 'number' then
        k = '"' .. k .. '"'
      end
      s = s .. '[' .. k .. '] = ' .. dump(v) .. ','
    end
    return s .. '} '
  else
    return tostring(o)
  end
end

-- table union: a += b
local function table_union(a, b)
  if type(a) ~= 'table' or type(b) ~= 'table' then
    error("Illegal argument: a or b are not a 'table' type")
  end
  for k, v in pairs(b) do
    if a[k] ~= nil then
      error(string.format("Duplicate keys detected:\nkey=%q, exist value=%q", dump(k), dump(a[k])))
      return
    end
    a[k] = v
  end
end

-- Define Config class
Config = { inner = {} }

function Config:new()
  local o = {}
  setmetatable(o, self)
  self.__index = self
  self.inner = {}
  return o
end

function Config:add(t)
  table_union(self.inner, t)
  return self
end

------------------------------------------------------
--               Wezterm configuration
------------------------------------------------------

local wzt = require('wezterm')
local act = wzt.action
local gui = wzt.gui

wzt.on('gui-startup', function(cmd)
  local tab, pane, window = wzt.mux.spawn_window(cmd or {})
  window:gui_window():set_inner_size(1600, 800)
  window:gui_window():set_position(200, 100)
  -- pane:split { size = 0.5 }  -- split 2 pane
end)

--------------------------------------- 启动项
local launch_menu = {
  default_prog = { 'nu' },
  launch_menu = {
    { label = '🟢 NuShell',            args = { 'nu' } },
    { label = '🟣 PowerShell Core',    args = { 'pwsh' } },
    { label = '🔵 Windows PowerShell', args = { 'powershell' } },
    { label = '⚪ Cmd',                 args = { 'cmd' } },
  },
}

--------------------------------------- 外观
-- local gpu_prefer = nil
-- for _, gpu in ipairs(gui.enumerate_gpus()) do
--   if gpu.backend == 'Vulkan' and gpu.device_type == 'DiscreteGpu' then
--     gpu_prefer = gpu
--     break
--   end
-- end

local colors = {
}

local appearance = {
  front_end = 'WebGpu',
  -- webgpu_preferred_adapter = gpu_prefer,
  animation_fps = 240,
  max_fps = 240,

  color_scheme_dirs = { 'e:/_config/wezterm/colors' },
  color_scheme = "retrowave",

  -- window_background_image = 'e:/_config/wezterm/bg_img.png',
  -- window_background_image_hsb = {
  --   -- Darken the background image by reducing it to 1/3rd
  --   brightness = 0.3,
  --
  --   -- You can adjust the hue by scaling its value.
  --   -- a multiplier of 1.0 leaves the value unchanged.
  --   hue = 1.0,
  --
  --   -- You can adjust the saturation also.
  --   saturation = 1.0,
  -- },
  --

  font = wzt.font { family = 'SpaceMono NFM' },
  font_size = 13.0,

  -- default_cursor_style = 'BlinkingBar',
  -- cursor_blink_rate = 333,

  window_decorations = 'NONE',
  window_background_opacity = 0.789,
  text_background_opacity = 0.3,
  win32_system_backdrop = 'Acrylic',
  inactive_pane_hsb = { saturation = 1.0, brightness = 0.5 },
  window_padding = { left = 0, right = 0, top = 0, bottom = 0 },

  use_fancy_tab_bar = false,
  tab_bar_at_bottom = true,
  enable_scroll_bar = false,
  hide_tab_bar_if_only_one_tab = false,
  tab_max_width = 9999,

  colors = {
    tab_bar = {
      background = '#0b0022',
      active_tab = {
        bg_color = '#2b2042',
        fg_color = '#FFFFFF',
        -- Specify whether you want "Half", "Normal" or "Bold" intensity for the
        intensity = 'Normal',
        -- Specify whether you want "None", "Single" or "Double" underline for
        underline = 'None',
        -- Specify whether you want the text to be italic (true) or not (false)
        italic = true,
        -- Specify whether you want the text to be rendered with strikethrough (true)
        strikethrough = false,
      },
      inactive_tab = {
        bg_color = '#1b1032',
        fg_color = '#808080',
      },
      inactive_tab_hover = {
        bg_color = '#3b3052',
        fg_color = '#909090',
        italic = true,
      },
      new_tab = {
        bg_color = '#1b1032',
        fg_color = '#808080',
        intensity = 'Bold',
      },
      new_tab_hover = {
        bg_color = '#3b3052',
        fg_color = '#909090',
        intensity = 'Bold',
      },
    },
  },
}

--------------------------------------- 快捷键
-- copy_mode
local copy_mode = gui.default_key_tables().copy_mode
table.insert(copy_mode,
  { key = 'h', mods = 'SHIFT', action = act.CopyMode 'MoveToStartOfLineContent' }
)
table.insert(copy_mode,
  { key = 'l', mods = 'SHIFT', action = act.CopyMode 'MoveToEndOfLineContent' }
)
table.insert(copy_mode,
  {
    key = 'y',
    mods = 'NONE',
    action = act.Multiple { act.CopyTo 'PrimarySelection', act.ClearSelection, act.CopyMode 'ClearSelectionMode' },
  }
)
-- search_mode
local search_mode = gui.default_key_tables().search_mode
table.insert(search_mode, { key = 'c', mods = 'CTRL', action = act.CopyMode 'Close' })

-- copy_mode <=> search_mode
table.insert(copy_mode,
  { key = 'i', mods = 'NONE', action = act.Search { CaseInSensitiveString = '' }, }
)
table.insert(search_mode, { key = 'Enter', mods = 'SHIFT', action = act.ActivateCopyMode })

local key_binding = {
  key_tables = {
    copy_mode = copy_mode,
    search_mode = search_mode,
  },
  keys = {
    { key = 'p',   mods = 'ALT',          action = act.ActivateCommandPalette },
    -- show debug overlay
    { key = 'F12', mods = 'NONE',         action = act.ShowDebugOverlay },
    { key = 'L',   mods = 'CTRL',         action = act.DisableDefaultAssignment },
    -- close tab
    { key = 'w',   mods = 'CTRL | SHIFT', action = act.CloseCurrentPane { confirm = false } },
    -- toggle window decorations
    {
      key = 'Enter',
      mods = 'ALT|SHIFT',
      ---@diagnostic disable-next-line: unused-local
      action = wzt.action_callback(function(win, pane)
        local overrides = win:get_config_overrides() or {}
        if overrides.window_decorations == 'NONE' then
          overrides.window_decorations = "TITLE | RESIZE"
        else
          overrides.window_decorations = "NONE"
        end
        -- will emit `window-config-reloaded` event
        win:set_config_overrides(overrides)
      end),
    },
    -- Pane zoom
    { key = ';',          mods = 'CTRL',       action = act.TogglePaneZoomState },
    -- Pane split
    { key = '=',          mods = 'ALT',        action = act.SplitHorizontal { domain = 'CurrentPaneDomain' } },
    { key = '-',          mods = 'ALT',        action = act.SplitVertical { domain = 'CurrentPaneDomain' } },
    -- Pane activate
    { key = 'LeftArrow',  mods = 'ALT',        action = act.ActivatePaneDirection 'Left' },
    { key = 'RightArrow', mods = 'ALT',        action = act.ActivatePaneDirection 'Right' },
    { key = 'UpArrow',    mods = 'ALT',        action = act.ActivatePaneDirection 'Up' },
    { key = 'DownArrow',  mods = 'ALT',        action = act.ActivatePaneDirection 'Down' },
    -- Pane resize
    { key = 'LeftArrow',  mods = 'ALT|SHIFT',  action = act.AdjustPaneSize { 'Left', 1 } },
    { key = 'RightArrow', mods = 'ALT|SHIFT',  action = act.AdjustPaneSize { 'Right', 1 } },
    { key = 'UpArrow',    mods = 'ALT|SHIFT',  action = act.AdjustPaneSize { 'Up', 1 } },
    { key = 'DownArrow',  mods = 'ALT|SHIFT',  action = act.AdjustPaneSize { 'Down', 1 } },
    -- Scroll
    { key = 'Home',       mods = 'SHIFT|CTRL', action = act.ScrollToTop },
    { key = 'End',        mods = 'SHIFT|CTRL', action = act.ScrollToBottom },
    { key = 'PageUp',     mods = 'SHIFT|CTRL', action = act.ScrollByPage(-1) },
    { key = 'PageDown',   mods = 'SHIFT|CTRL', action = act.ScrollByPage(1) },
    { key = 'UpArrow',    mods = 'SHIFT|CTRL', action = act.ScrollByLine(-1) },
    { key = 'DownArrow',  mods = 'SHIFT|CTRL', action = act.ScrollByLine(1) },
    { key = 'LeftArrow',  mods = 'SHIFT|CTRL', action = act.ScrollToPrompt(-1) },
    { key = 'RightArrow', mods = 'SHIFT|CTRL', action = act.ScrollToPrompt(1) },
    -- QuickSelect
    { key = 'f',          mods = 'SHIFT|CTRL', action = act.QuickSelect },
    -- Show
    { key = 'u',          mods = 'ALT',        action = act.ShowLauncherArgs { title = '🔭 Tabs', flags = 'FUZZY|TABS' } },
    {
      key = 'i',
      mods = 'ALT',
      action = act.ShowLauncherArgs { title = '🔭 快捷键/命令', flags = 'FUZZY|KEY_ASSIGNMENTS|COMMANDS' }
    },
    {
      key = 'o',
      mods = 'ALT',
      action = act.ShowLauncherArgs { title = '🔭 启动项/域/工作区', flags =
      'LAUNCH_MENU_ITEMS|DOMAINS|WORKSPACES' }
    },
  }
}

return Config:new()
    :add(launch_menu)
    :add(appearance)
    :add(key_binding)
    :add(colors)
    .inner

```

**Comment by lost22git** [comment]
No, I just thought maybe the author wasn't too interested in it

**Comment by wez** [comment]
> No, I just thought maybe the author wasn't too interested in it

I saw this issue and thought "weird, because I thought it worked for me last time I was on Windows", but since I was laid off I no longer use Windows for work, and I now generally only fire up my Windows system when I have enough time and headspace to play games, I just haven't had time to look into this.

The reality is that there's an increasing backlog of issues: wezterm is popular and people are generating issues faster than I can read and resolve them, and I'm falling behind on them. Triaging bugs is not an energizing activity; it's work.

Since I was part of the mass of tech industry layoffs, I am temporarily in a position to consider making wezterm a source of income and treat it as a job, but, while there are some sponsors, there are not currently enough to make that sustainable. I posted about this over here: https://fosstodon.org/@wez/110132381130728314
If everyone that starred wezterm sponsored at $1/month, things start to get interesting.

With regard to this issue: I'm happy for it to remain open to remind me to look into it. But I'm also not going to lose sleep over it being closed if you are frustrated that I haven't looked at it. 

For the sake of transparency, I don't know when I'll have time to look into it.

Also for the sake of more transparency: I don't offer any kind of warranty or SLA to sponsors in case you or any other reader of this comment is interpreting it as me trying to solicit a transaction. wezterm is provided as-is with no warranty, as explained in the license, regardless of whether someone is a sponsor or not.
